### ISSUE TITLE
Remote externalities key downloading slows down at the end

### ISSUE THREAD

**Issue by ggwpez** [paritytech/polkadot-sdk#2096]
Somehow the remote externalities get slower at the end. This was determined to not be related to RPC rate limits but seems to be a problem of the externalities.  
A debug flag like `--measure-kv-timing output.json` could be added to measure and investigate the KV timings later on.

Example: normally around 12-14 kV/s, it slows down to 200-500 V/s. This was also noticed by @liamaharon.

![image (1)](https://github.com/paritytech/polkadot-sdk/assets/10380170/0e28227b-475b-4ad7-bb43-187d073dbe1d)

(Marking as mentor since adding the flag should be simple=

**Comment by eagr** [comment]
I noticed the related code is deprecated in favor of the standalone [try-runtime-cli](https://github.com/paritytech/try-runtime-cli). Do you still want to do this here? @ggwpez @liamaharon

**Comment by liamaharon** [comment]
> I noticed the related code is deprecated in favor of the standalone [try-runtime-cli](https://github.com/paritytech/try-runtime-cli). Do you still want to do this here? @ggwpez @liamaharon

This issue originates in `remote-externalities`, which `try-runtime-cli` depends on. 

Same crate you are adding the parallel scraping to :)

**Comment by eagr** [comment]
Yeah, that part of the change definitely goes this repo. I mean the cli flag part. Do you still want to do that here? :))

**Comment by liamaharon** [comment]
> Yeah, that part of the change definitely goes this repo. I mean the cli flag part. Do you still want to do that here? :))

Seems it needs an issue in both repos. One to add the feature to remote-ext, and another to use it in try-runtime-cli
